src/features/navigator/NarrativeList/NarrativeItemDropdown.tsx

```diff
@@ -0,0 +1,116 @@
+import { FC } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { FontAwesomeIcon as FAIcon } from '@fortawesome/react-fontawesome';
+import { faCheck, faCaretDown } from '@fortawesome/free-solid-svg-icons';
+import { useAppSelector } from '../../../common/hooks';
+import { Dropdown } from '../../../common/components/Dropdown';
+import { SelectOption } from '../../../common/components/Select';
+import { keepParamsForLocation, narrativePath, searchParams } from '../common';
+import { categorySelected, navigatorSelected } from '../navigatorSlice';
+import classes from './NarrativeList.module.scss';
+
+type NarrativeItemDropdownProps = {
+  narrative: string;
+  onVersionSelect: (e: number) => void;
+  version: number;
+};
+
+const NarrativeItemDropdown: FC<NarrativeItemDropdownProps> = ({
+  narrative,
+  onVersionSelect,
+  version,
+}) => {
+  const narrativeSelected = useAppSelector(navigatorSelected);
+  console.log({ narrativeSelected }); // eslint-disable-line no-console
+  const categorySet = useAppSelector(categorySelected);
+  const navigate = useNavigate();
+  const [id, obj, ver] = narrative.split('/');
+  const versionLatest = +ver;
+  const [idSelected, objSelected, versionSelected] = narrativeSelected
+    ? narrativeSelected.split('/')
+    : [null, null, null];
+  const active = idSelected === id && objSelected === obj;
+  console.log({ active, narrativeSelected }); // eslint-disable-line no-console
+  const loc = useLocation();
+  const keepSearch = keepParamsForLocation({
```

Comment:
Sometimes you want to say "link to `/public`" but keep the sort and search filters. Since the search and sort are managed with query parameters, it is desirable to preserve their values in some way, and that is the purpose of these functions.

Reply:
Let's write an explicit `generatePathWithSearchParameters` function instead. Look into using redux to store query parameter state.